### PR TITLE
Pr/ping 2156

### PR DESCRIPTION
**Pull request by febrimaulana**
### Summary by CodeRabbit

- Test: Updated unit tests for `Comment`, `Footer`, `ContainerComment`, `ContainerReply`, `ReplyComment`, `HeaderFeed`, `RenderItem`, and `calculateTime` to reflect changes in the UI and functionality.
- New Feature: Added a new prop `isShowDM` to the `RenderListFeed` component in `TopicPageScreen` for enhanced user interaction (details of usage not provided in the diff).
- Chore: Added Jest mock for the `@react-native-clipboard/clipboard` module in `HeaderFeed.test.js` for improved testing.

### CODE REVIEW

**Comment by bastian-purrer** [comment]
Image description   CodeRabbit



Currently reviewing new changes in this PR...

<details>
<summary>Commits</summary>
Files that changed from the base of the PR and between c7a8c9bc60ce1c34b647343a19718cfc156d4ce8 and e5673c5340113754eab4f87af4f9422332d328db commits.
</details>

<details>
<summary>Files selected (7)</summary>

* __tests__/components/Comments/Comment.test.js (8)
* __tests__/components/Comments/ContainerComment.test.js (6)
* __tests__/components/Footer/Footer.test.js (2)
* __tests__/screens/FeedScreen/HeaderFeed.test.js (1)
* __tests__/screens/News/RenderItem.test.js (1)
* __tests__/utils/time.test.js (2)
* src/screens/TopicPageScreen/MemoizedListComponent.js (1)
</details>






---

File: src/screens/TopicPageScreen/MemoizedListComponent.js [hunk 1]
```diff
@@ -166,6 +166,7 @@ const RenderListFeed = (props) => {
             isSelf={item.anonimity ? false : userId === item?.actor?.id}
             onPressScore={showScoreAlertDialog}
             showScoreButton={showScoreButton}
+            isShowDM
```
Comment (bastian-purrer) [inline src/screens/TopicPageScreen/MemoizedListComponent.js:169]: Image description   CodeRabbit

The new prop `isShowDM` is being passed to the `RenderListFeed` component, but it's not clear where this prop is coming from. If it's a new prop for the parent component, make sure it's properly defined and passed down. If it's supposed to be derived from some other state or props, ensure that logic is implemented correctly.
```diff
-             isShowDM
+             isShowDM={someConditionOrProp}
```
Please replace `someConditionOrProp` with the actual condition or prop you want to pass as `isShowDM`.

File: __tests__/components/Comments/ContainerComment.test.js [hunk 2]
```diff
@@ -138,7 +138,8 @@ describe('ContainerComment should run correctly', () => {
     const findCommentAndUpdate = jest.fn();
 
     const contextValue = {
-      feeds: [feedsState, () => jest.fn()]
+      feeds: [feedsState, () => jest.fn()],
+      profile: [{}, () => jest.fn()]
```
Comment (bastian-purrer) [inline __tests__/components/Comments/ContainerComment.test.js:142]: Image description   CodeRabbit

The context value now includes a `profile` prop. Ensure that all components using this context are updated to handle the new structure of the context value.
```diff
-      feeds: [feedsState, () => jest.fn()]
+      feeds: [feedsState, () => jest.fn()],
+      profile: [{}, () => jest.fn()]
```

File: __tests__/components/Comments/ContainerComment.test.js [hunk 3]
```diff
@@ -164,7 +165,8 @@ describe('ContainerComment should run correctly', () => {
...
```
Comment (bastian-purrer) [inline __tests__/components/Comments/ContainerComment.test.js:169]: Image description   CodeRabbit

Same as above, ensure that all components using this context are updated to handle the new structure of the context value.
```diff
-      feeds: [feedsState, () => jest.fn()]
+      feeds: [feedsState, () => jest.fn()],
+      profile: [{}, () => jest.fn()]
```

File: __tests__/components/Comments/ContainerComment.test.js [hunk 4]
```diff
@@ -203,9 +205,11 @@ describe('ContainerComment should run correctly', () => {
...
   it('ContainerReply should match snapshot', () => {
     const {toJSON, getAllByTestId} = render(
-      <ContainerReply>
-        <View testID="children" />{' '}
-      </ContainerReply>
+      <Context.Provider value={{profile: [{}, () => jest.fn()]}}>
+        <ContainerReply>
+          <View testID="children" />{' '}
+        </ContainerReply>
+      </Context.Provider>
```
Comment (bastian-purrer) [inline __tests__/components/Comments/ContainerComment.test.js:212]: Image description   CodeRabbit

The `ContainerReply` component is now wrapped in a `Context.Provider` with a `profile` prop. Make sure that the `ContainerReply` component can handle this new prop and that it doesn't break any existing functionality.
```diff
-      <ContainerReply>
-        <View testID="children" />{' '}
-      </ContainerReply>
+      <Context.Provider value={{profile: [{}, () => jest.fn()]}}>
+        <ContainerReply>
+          <View testID="children" />{' '}
+        </ContainerReply>
+      </Context.Provider>
```

File: __tests__/components/Comments/ContainerComment.test.js [hunk 5]
```diff
@@ -218,7 +222,8 @@ describe('ContainerComment should run correctly', () => {
     const findCommentAndUpdate = jest.fn();
 
     const contextValue = {
-      feeds: [feedsState, () => jest.fn()]
+      feeds: [feedsState, () => jest.fn()],
+      profile: [{}, () => jest.fn()]
```
Comment (bastian-purrer) [inline __tests__/components/Comments/ContainerComment.test.js:226]: Image description   CodeRabbit

Again, ensure that all components using this context are updated to handle the new structure of the context value.
```diff
-      feeds: [feedsState, () => jest.fn()]
+      feeds: [feedsState, () => jest.fn()],
+      profile: [{}, () => jest.fn()]
```

File: __tests__/components/Comments/ContainerComment.test.js [hunk 6]
```diff
@@ -246,13 +251,15 @@ describe('ContainerComment should run correctly', () => {
...
-      />
+      <Context.Provider value={{profile: [{}, () => jest.fn()]}}>
+        <ReplyComment
+          countComment={comments2[0].children_counts.comment}
+          indexFeed={indexFeed}
+          findCommentAndUpdate={findCommentAndUpdate}
+          navigateToReplyView={navigateToReplyView}
+          data={comments2[0].latest_children.comment}
+        />
+      </Context.Provider>
```
Comment (bastian-purrer) [inline __tests__/components/Comments/ContainerComment.test.js:262]: Image description   CodeRabbit

The `ReplyComment` component is now wrapped in a `Context.Provider` with a `profile` prop. Make sure that the `ReplyComment` component can handle this new prop and that it doesn't break any existing functionality.
```diff
-      <ReplyComment
-        countComment={comments2[0].children_counts.comment}
-        indexFeed={indexFeed}
-        findCommentAndUpdate={findCommentAndUpdate}
-        navigateToReplyView={navigateToReplyView}
-        data={comments2[0].latest_children.comment}
-      />
+      <Context.Provider value={{profile: [{}, () => jest.fn()]}}>
+        <ReplyComment
+          countComment={comments2[0].children_counts.comment}
+          indexFeed={indexFeed}
+          findCommentAndUpdate={findCommentAndUpdate}
+          navigateToReplyView={navigateToReplyView}
+          data={comments2[0].latest_children.comment}
+        />
+      </Context.Provider>
```

File: __tests__/components/Comments/ContainerComment.test.js [hunk 7]
```diff
@@ -262,13 +269,15 @@ describe('ContainerComment should run correctly', () => {
...
```
Comment (bastian-purrer) [inline __tests__/components/Comments/ContainerComment.test.js:280]: Image description   CodeRabbit

Same as above, ensure that the `ReplyComment` component can handle the new `profile` prop and that it doesn't break any existing functionality.
```diff
-      <ReplyComment
-        countComment={comments2[0].children_counts.comment}
-        indexFeed={indexFeed}
-        findCommentAndUpdate={findCommentAndUpdate}
-        navigateToReplyView={navigateToReplyView}
-        data={comments2[0].latest_children.comment}
-      />
+      <Context.Provider value={{profile: [{}, () => jest.fn()]}}>
+        <ReplyComment
+          countComment={comments2[0].children_counts.comment}
+          indexFeed={indexFeed}
+          findCommentAndUpdate={findCommentAndUpdate}
+          navigateToReplyView={navigateToReplyView}
+          data={comments2[0].latest_children.comment}
+        />
+      </Context.Provider>
```

**Review by bastian-purrer — COMMENTED** [review]
Image description   CodeRabbit

<details>
<summary>Commits</summary>
Files that changed from the base of the PR and between c7a8c9bc60ce1c34b647343a19718cfc156d4ce8 and e5673c5340113754eab4f87af4f9422332d328db commits.
</details>

<details>
<summary>Files selected (7)</summary>

* __tests__/components/Comments/Comment.test.js (8)
* __tests__/components/Comments/ContainerComment.test.js (6)
* __tests__/components/Footer/Footer.test.js (2)
* __tests__/screens/FeedScreen/HeaderFeed.test.js (1)
* __tests__/screens/News/RenderItem.test.js (1)
* __tests__/utils/time.test.js (2)
* src/screens/TopicPageScreen/MemoizedListComponent.js (1)
</details>







<details>
<summary>Files skipped from review due to trivial changes (5)</summary>

* __tests__/components/Comments/Comment.test.js
* __tests__/components/Footer/Footer.test.js
* __tests__/screens/FeedScreen/HeaderFeed.test.js
* __tests__/screens/News/RenderItem.test.js
* __tests__/utils/time.test.js

</details>

<details>
<summary>Review comments generated (7)</summary>

* Review: 7
* LGTM: 0

</details>

---

<details>
<summary>Tips</summary>

### Chat with Image description  CodeRabbit Bot (`@coderabbitai`)
- Reply on review comments left by this bot to ask follow-up questions. A review comment is a comment on a diff or a file.
- Invite the bot into a review comment chain by tagging `@coderabbitai` in a reply.

### Code suggestions
- The bot may make code suggestions, but please review them carefully before committing since the line number ranges may be misaligned. 
- You can edit the comment made by the bot and manually tweak the suggestion if it is slightly off.

### Pausing incremental reviews
- Add `@coderabbitai: ignore` anywhere in the PR description to pause further reviews from the bot.

</details>

**Review by asaadam — APPROVED** [review]
Approved!

**Pull request merged**